teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -0,0 +1,680 @@
+  /foo-task
+    /0.1
+    /0.2 
+  /bar-task
+    /0.1
+    /0.2
+    /0.3
+  /...
+./pipeline/
+  /go-release
+    /0.1
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /OWNERS
+  /foo-pipeline
+    /0.1
+    /0.2 
+  /bar-pipeline
+    /0.1
+    /0.2
+    /0.3
+  /...
+```
+
+For further information, see [TEP-0003: Tekton Catalog Organization][tep-0003].
+
+The current directory-based organization and versioning presents challenges:
+- Bumping versions of resources is cumbersome which encourages making "minor" changes in existing versions. Some changes
+that seem minor cause surprising breaking changes for users, such as described in [tektoncd/catalog#784][784].
+- Resources that are frequently updated with new versions end up with a lot of duplications because any change requires
+copying over everything into a new directory. For example, [git-clone][git-clone] `Task` has seven versions with a lot
+of duplication across versions.
+
+Previously, there was tight coupling between the organization and how users referenced resources which made it hard to 
+change the organization of the Catalog without breaking users. In [TEP-0110][tep-0110], we decoupled the Catalog 
+organization from resource reference. Now, we can create a new organization and versioning contract for Catalogs to
+address the challenges with the directory-based organization and versioning.
+
+The aim of this TEP is to create a new organization and versioning for Tekton Catalogs, building on the prior work and 
+experiences from [TEP-0003][tep-0003]. In [TEP-0079: Tekton Catalog Support Tiers][tep-0079], we plan to add a Tekton 
+Official Catalog which will use this new organization and versioning contract - therefore, this proposal is a blocker 
+for [TEP-0079][tep-0079].
+
+## Definitions
+
+Terms used in this TEP are defined as follows:
+
+* **Repository**: A versioned source of a project. There's no special definition of "repository" in the context of this
+TEP. A repository may be a Catalog.
+
+* **Catalog**: A repository that complies with the organization contract defined by Tekton. The directory-based contract
+is defined in [TEP-0003: Tekton Catalog Organization][tep-0003-org], and [git-based contract](#organization-contract) is
+defined in this TEP. A Catalog may contain one or more resources.
+
+* **Resource**: Item shared in a Catalog e.g. `Task` or `Pipeline`.
+
+* **Hub**: User interface for one or more Catalogs.
+
+## Proposal
+
+We propose supporting git-based versioning of Tekton Catalogs.
+
+### Git-Based Versioning
+
+#### Organization Contract
+
+We propose removing the directories - `0.1`...`0.n` - that were used for versioning resources in the Catalogs. The new
+organization contract with git-based versioning would be:
+
+```bash
+# Optional: owner(s) and reviewer(s) of the resource (all versions)
+./{resource-type}/{resource-name}/OWNERS
```

Comment:
We can still have owners in here, but it's important to note that owners of a specific resource won't be able anymore to create a release of their own resource. That's ok, but I think it's a change we should highlight.

Reply:
added a note about that, thank you!